nupic/research/frameworks/continuous_learning/mnist_data_utils.py

```diff
@@ -0,0 +1,151 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import os
+import tempfile
+
+import torch
+from torch.utils.data import DataLoader
+from torchvision import transforms
+
+from nupic.research.frameworks.pytorch.dataset_utils import PreprocessedDataset
+
+
+def mnist_classwise_loader(
+    data_dir="/home/ec2-user/nta/data/mnist",
+    transform=None,
+    batch_size=32,
+):
+    """ Create DataLoader instances for the MNIST train and test sets.
+        :param data_dir: Where your MNIST data lives.
+        :param transform: Any transforms you'd like to apply to your data.
+         Defaults to None.
+        :param batch_size: Desired batch_size
+    """
+    std_transform = transforms.Lambda(lambda x: (x[0].float(), x[1].long()))
+    if transform is not None:
+        transform = transforms.Compose([
+            std_transform,
+            transform,
+        ])
+
+    else:
+        transform = std_transform
+
+    train_loader, test_loader = [], []
+    num_classes = 10
+
+    for class_ in range(num_classes):
+        train_dataset = ClasswiseDataset(
+            cachefilepath=data_dir,
+            basename="mnist_train_",
+            qualifiers=[class_],
+            transform=transform,
+        )
+
+        train_loader.append(
+            DataLoader(
+                train_dataset, batch_size=batch_size, shuffle=False, drop_last=True
+            )
+        )
+
+        test_dataset = ClasswiseDataset(
+            cachefilepath=data_dir,
+            basename="mnist_test_",
+            qualifiers=[class_],
+            transform=transform,
+        )
+
+        test_loader.append(
+            DataLoader(
+                test_dataset, batch_size=batch_size, shuffle=False, drop_last=True
```

Comment:
I think batch_size can be much higher for testing, which would make validation go faster.